interpreter/prune.go

```diff
@@ -65,13 +66,17 @@ type astPruner struct {
 // compiled and constant folded expressions, but is not willing to constant
 // fold(and thus cache results of) some external calls, then they can prepare
 // the overloads accordingly.
-func PruneAst(expr *exprpb.Expr, state EvalState) *exprpb.Expr {
+func PruneAst(expr *exprpb.Expr, macroCalls map[int64]*exprpb.Expr, state EvalState) *exprpb.ParsedExpr {
```

Comment:
The return type change here might break consumers, but looks like this is a required modification (probably worth mentioning in the release notes).

Reply:
Agreed. this is a breaking change if anyone is currently using this interface. I'll make sure it's in the release notes.